README.md
# RiskPortfolios
Computation of risk-based portfolios in R

`RiskPortfolios` ([Ardia et al., 2017](http://dx.doi.org/10.21105/joss.00171)) is an R package for constructing risk-based portfolios dedicated to portfolio managers 
and quantitative analysts. It provides a set of functionalities to build mean-variance, minimum variance, inverse-volatility weighted, 
equal-risk-contribution, maximum diversification, and risk-efficient portfolios. As risk-based portfolios are
mainly based on covariances, the package also provides a large set of covariance matrix estimators. See [Ardia et al. (2017)](http://dx.doi.org/10.21105/joss.00171) for details. A Monte Carlo study relying on `RiskPortfolios` is presented in [Ardia et al. (2017)](http://dx.doi.org/10.1007/s10479-017-2474-7).

The latest stable version of `RiskPortfolios` is available at [https://cran.r-project.org/package=RiskPortfolios](https://cran.r-project.org/package=RiskPortfolios).

The latest development version of `RiskPortfolios` is available at [https://github.com/ArdiaD/RiskPortfolios](https://github.com/ArdiaD/RiskPortfolios).

Please cite `RiskPortfolios` in publications:

Ardia, D., Bolliger, G., Boudt, K., Gagnon-Fleury, J.-P. (2017).      
The impact of covariance misspecification in risk-based portfolios.      
_Annals of Operations Research_ **254**(1--2), pp 1-16.      
[http://dx.doi.org/10.1007/s10479-017-2474-7](http://dx.doi.org/10.1007/s10479-017-2474-7)   
[http://dx.doi.org/10.2139/ssrn.2650644](http://dx.doi.org/10.2139/ssrn.2650644) 
   
Ardia, D., Boudt, K., Gagnon-Fleury, J.-P. (2017).      
RiskPortfolios: Computation of risk-based portfolios in R.        
_Journal of Open Source Software_ **10**(2).    
[http://dx.doi.org/10.21105/joss.00171](http://dx.doi.org/10.21105/joss.00171)      

NEWS
Changes in Version 2.1.5 (DA)
  o Update references
  o DOI fixed
  
Changes in Version 2.1.4 (DA)
  o Fix description and citation

Changes in Version 2.1.3 (DA)
  o Fix description and citation

Changes in Version 2.1.2 (DA)
  o References updated

Changes in Version 2.1.2 (DA)
  o Gradient added
  o Max decorrelation portfolio added

Changes in Version 2.1.0 (DA)
  o CRAN release
  o documentation updated
  o THANKS file added

Changes in Version 2.00.11 (DA)
  o Documentation improved and finalized

Changes in Version 2.00.10 (DA)
  o LB and UB added properly
  o doc modified

Changes in Version 2.00.09 (DA)
  o dataset added
  o doc RiskPortfolios added

Changes in Version 2.00.08 (DA)
  o tests added
  o doc fixed
  o positivity fixed

Changes in Version 2.00.07 (DA)
  o Roxygen documentation
  o testthat added

Changes in Version 2.00.06 (DA)
  o update CITATION

Changes in Version 2.00.05 (DA)
  o update informations and references

Changes in Version 2.00.04 (DA)
  o serveral error fixes
  o new Imports instead of Depends

Changes in Version 2.00.03 (DA)
  o erc initialized at 1/sigma normlized

Changes in Version 2.00.02 (DA)
  o w0 starting value can now be passed in controls
  o references updated

Changes in Version 2.00.01 (DA)
  o Renaming of the package with emphasis of risk-based portfolios
  o Added inverse volatility portfolio

Changes in Version 1.02.02 (DA)
  o DESCRIPTION file adapted to new standard

Changes in Version 1.02.01 (DA)
  o major revision of the package

Changes in Version 1.01.02 (DA)
  o bug fixd thanks to Samo Pahor

Changes in Version 1.01.01 (DA)
  o package's name
  o package's version

Changes in Version 1-00.01 (DA)
  o first release

tests/testthat/test_portfolio.R
context("Portfolios") 

test_that("Portfolios", { 
  T = 100
  N = 25
  set.seed(123)
  rets  = matrix(rnorm(T * N), nrow = T, ncol = N)
  
  f.test.pos = function(W) {
    out = all(w >= 0)
    return(out)
  }
  
  f.test.sum = function(w) {
    out = abs(sum(w) - 1) < 1e-4
    return(out)
  }
  
  mu = meanEstimation(rets)
  Sigma = covEstimation(rets)
  semiDev = semidevEstimation(rets)
  LB = rep(0.02, N)
  UB = rep(0.95, N)
  
  w = optimalPortfolio(mu = mu, Sigma = Sigma)
  expect_equal(length(w), N)
  
  w = optimalPortfolio(mu = mu, Sigma = Sigma, 
                       control = list(gamma = 1))
  expect_equal(length(w), N)
  
  w = optimalPortfolio(mu = mu, Sigma = Sigma, 
                       control = list(type = 'mv'))
  expect_equal(length(w), N)
  
  w = optimalPortfolio(mu = mu, Sigma = Sigma, 
                       control = list(type = 'mv', constraint = 'none'))
  expect_equal(length(w), N)
  
  w = optimalPortfolio(mu = mu, Sigma = Sigma, 
                       control = list(type = 'mv', constraint = 'lo'))
  expect_equal(length(w), N)
  expect_true(f.test.pos(w))
  expect_true(f.test.sum(w))
  
  w = optimalPortfolio(mu = mu, Sigma = Sigma, 
                       control = list(type = 'mv', constraint = 'user', LB = LB, UB = UB))
  expect_equal(length(w), N)
  expect_true(f.test.sum(w))
  
  w = optimalPortfolio(mu = mu, Sigma = Sigma, 
                       control = list(type = 'mv', constraint = 'gross'))
  expect_equal(length(w), N)
  expect_true(f.test.sum(w))
  
  w = optimalPortfolio(mu = mu, Sigma = Sigma, 
                       control = list(type = 'mv', constraint = 'gross', LB = LB, UB = UB))
  expect_equal(length(w), N)
  expect_true(f.test.sum(w))
  
  w = optimalPortfolio(mu = mu, Sigma = Sigma, 
                       control = list(type = 'mv', constraint = 'gross', gross.c = 1.2))
  expect_equal(length(w), N)
  expect_true(f.test.sum(w))
  
  w = optimalPortfolio(Sigma = Sigma, 
                       control = list(type = 'minvol'))
  expect_equal(length(w), N)
  expect_true(f.test.sum(w))
  
  w = optimalPortfolio(Sigma = Sigma, 
                       control = list(type = 'minvol', constraint = 'none'))
  expect_equal(length(w), N)
  expect_true(f.test.sum(w))
  
  w = optimalPortfolio(Sigma = Sigma, 
                       control = list(type = 'minvol', constraint = 'lo'))
  expect_equal(length(w), N)
  expect_true(f.test.pos(w))
  expect_true(f.test.sum(w))
  
  w = optimalPortfolio(Sigma = Sigma, 
                       control = list(type = 'minvol', constraint = 'user', LB = LB, UB = UB))
  expect_equal(length(w), N)
  expect_true(f.test.sum(w))
  
  w = optimalPortfolio(Sigma = Sigma, 
                       control = list(type = 'minvol', constraint = 'gross'))
  expect_equal(length(w), N)
  expect_true(f.test.sum(w))
  
  w = optimalPortfolio(Sigma = Sigma, 
                       control = list(type = 'minvol', constraint = 'gross', gross.c = 1.2))
  expect_equal(length(w), N)
  expect_true(f.test.sum(w))
  
  
  w = optimalPortfolio(Sigma = Sigma, 
                       control = list(type = 'invvol'))
  expect_equal(length(w), N)
  expect_true(f.test.pos(w))
  expect_true(f.test.sum(w))
  
  w = optimalPortfolio(Sigma = Sigma, 
                       control = list(type = 'erc'))
  expect_equal(length(w), N)
  expect_true(f.test.sum(w))
  
  w = optimalPortfolio(Sigma = Sigma, 
                       control = list(type = 'erc', constraint = 'lo'))
  expect_equal(length(w), N)
  expect_true(f.test.pos(w))
  expect_true(f.test.sum(w))
  
  w = optimalPortfolio(Sigma = Sigma, 
                       control = list(type = 'erc', constraint = 'user', LB = LB, UB = UB))
  expect_equal(length(w), N)
  expect_true(f.test.sum(w))
  
  w = optimalPortfolio(Sigma = Sigma, 
                       control = list(type = 'maxdiv'))
  expect_equal(length(w), N)
  expect_true(f.test.sum(w))
  
  w = optimalPortfolio(Sigma = Sigma, 
                       control = list(type = 'maxdiv', constraint = 'lo'))
  expect_equal(length(w), N)
  expect_true(f.test.pos(w))
  expect_true(f.test.sum(w))
  
  w = optimalPortfolio(Sigma = Sigma, 
                       control = list(type = 'maxdiv', constraint = 'user', LB = LB, UB = UB))
  expect_equal(length(w), N)
  expect_true(f.test.sum(w))
  
  w = optimalPortfolio(Sigma = Sigma, semiDev = semiDev, 
                       control = list(type = 'riskeff'))
  expect_equal(length(w), N)
  expect_true(f.test.sum(w))
  
  w = optimalPortfolio(Sigma = Sigma, semiDev = semiDev, 
                       control = list(type = 'riskeff', constraint = 'lo'))
  expect_equal(length(w), N)
  expect_true(f.test.pos(w))
  expect_true(f.test.sum(w))
  
  w = optimalPortfolio(Sigma = Sigma, semiDev = semiDev, 
                       control = list(type = 'riskeff', constraint = 'user', LB = LB, UB = UB))
  expect_equal(length(w), N)
  expect_true(f.test.pos(w))
  expect_true(f.test.sum(w))
})

R/meanEstimation.R
#' @name meanEstimation
#' @aliases meanEstimation
#' @title Estimation of mean returns
#' @description Function which is used to compute the estimation of the mean returns.
#' @details The argument \code{control} is a list that can supply any of the following
#' components:
#' \itemize{ 
#' \item \code{type} method used to estimate the mean returns,
#' among \code{'naive'}, \code{'ewma'}, \code{'bs'} and \code{'mart'} where:
#' 
#' \code{'naive'} is used to compute the arithmetic mean of the returns.
#' 
#' \code{'ewma'} is used to compute the exponential weighted moving average
#' mean of the returns. The data must be sorted from the oldest to the latest. See RiskMetrics (1996).
#' 
#' \code{'bs'} is used to compute the Bayes-Stein estimation. See Jorion (1986).
#' 
#' \code{'mart'} is used to compute the Martinelli (2008) implied returns.
#' 
#' Default: \code{type = 'naive'}.
#' 
#' \item \code{lambda} decay parameter. Default: \code{lambda = 0.94}.
#' }
#' 
#' @param rets a \eqn{(T \times N)}{(T x N)} matrix of past returns.
#' @param control control parameters (see *Details*).
#' @return A \eqn{(N \times 1)}{(N x 1)} vector of expected returns.
#' @author David Ardia, Kris Boudt and Jean-Philippe Gagnon Fleury.
#' @references 
#' Jorion, P. (1986).  
#' Bayes-Stein estimation for portfolio analysis. 
#' \emph{Journal of Finance and Quantitative Analysis} \bold{21}(3), pp.279-292. 
#' 
#' Martellini, L. (2008).  
#' Towards the design of better equity benchmarks.
#' \emph{Journal of Portfolio Management} \bold{34}(4), Summer,pp.34-41. 
#' 
#' RiskMetrics (1996)
#' \emph{RiskMetrics Technical Document}.
#' J. P. Morgan/Reuters. 
#' \url{https://www.msci.com/documents/10199/5915b101-4206-4ba0-aee2-3449d5c7e95a}
#' @keywords htest
#' @examples
#' # Load returns of assets or portfolios
#' data("Industry_10")
#' rets = Industry_10
#' 
#' # Naive estimation of the mean
#' meanEstimation(rets)
#' 
#' # Naive estimation of the mean
#' meanEstimation(rets, control = list(type = 'naive'))
#' 
#' # Ewma estimation of the mean with default lambda = 0.94
#' meanEstimation(rets, control = list(type = 'ewma'))
#' 
#' # Ewma estimation of the mean with lambda = 0.9
#' meanEstimation(rets, control = list(type = 'ewma', lambda = 0.9))
#' 
#' # Martinelli's estimation of the mean
#' meanEstimation(rets, control = list(type = 'mart'))
#' 
#' # Bayes-Stein's estimation of the mean
#' meanEstimation(rets, control = list(type = 'bs'))
#' @export
meanEstimation <- function(rets, control = list()) {
  
  if (missing(rets)) {
    stop("rets is missing")
  }
  if (!is.matrix(rets)) {
    stop("rets must be a (T x N) matrix")
  }
  
  ctr <- .ctrMean(control)
  
  if (ctr$type[1] == "naive") {
    mu <- .naiveMean(rets = rets)
  } else if (ctr$type[1] == "ewma") {
    mu <- .ewmaMean(rets = rets, lambda = ctr$lambda)
  } else if (ctr$type[1] == "bs") {
    mu <- .bsMean(rets = rets)
  } else if (ctr$type[1] == "mart") {
    mu <- .martMean(rets = rets)
  } else {
    stop("control$type is not well defined")
  }
  return(mu)
}

.ctrMean <- function(control = list()) {
  ## Function used to control the list input INPUTS control : a control
  ## ist The argument control is a list that can supply any of the
  ## following components type : default = 'naive' lambda : default = 0.94
  ## OUTPUTs control : a list
  if (!is.list(control)) {
    stop("control must be a list")
  }
  if (length(control) == 0) {
    control <- list(type = "naive", lambda = 0.94)
  }
  nam <- names(control)
  if (!("type" %in% nam) || is.null(control$type)) {
    control$type <- c("naive", "ewma", "bs", "mart")
  }
  if (!("lambda" %in% nam) || is.null(control$lambda)) {
    control$lambda <- 0.94
  }
  return(control)
}

.naiveMean <- function(rets) {
  ## Compute the naive mean INPUTs rets : matrix (T x N) returns OUTPUTs
  ## mu : vector (N x 1) mean
  mu <- colMeans(rets)
  return(mu)
}

.ewmaMean <- function(rets, lambda) {
  ## Compute the exponential weighted moving average INPUTs rets : matrix
  ## (T x N) returns lambda : scalar OUTPUTs mu : vector (N x 1) mean NOTE
  ## the dates should be in ascending order, i.e. ordest at the top and
  ## newest at the bottom
  t <- dim(rets)[1]
  w <- lambda^(t:1)
  w <- w/sum(w)
  w <- matrix(data = w, nrow = t, ncol = dim(rets)[2], byrow = FALSE)
  mu <- colSums(w * rets)
  return(mu)
}

.bsMean <- function(rets) {
  ## Compute the Bayes-Stein mean INPUTs rets : matrix (T x N) returns
  ## OUTPUTs mu : vector (N x 1) mean !!! TOFIX !!! estime pour l'instant
  ## une matrice de var-cov de facon standard verifier avec david si on
  ## laisse le choix en input
  mu <- colMeans(rets)
  Sigma <- cov(rets)
  invSigma <- solve(Sigma)
  N <- dim(Sigma)[1]
  T <- length(mu)
  i <- rep(1, N)
  invSigmai <- crossprod(invSigma, i)
  w_min <- (invSigmai)/as.numeric(crossprod(i, invSigmai))
  mu_min <- crossprod(mu, w_min)
  invSigmaMu <- crossprod(invSigma, mu - mu_min)
  phi <- (N + 2)/((N + 2) + T * crossprod(mu - mu_min, invSigmaMu))
  phi <- max(min(phi, 1), 0)
  mu <- (1 - phi) * mu + phi * mu_min
  return(mu)
}

.martMean <- function(rets) {
  ## Compute the Martinelli mean INPUTs rets : matrix (T x N) returns
  ## OUTPUTs mu : vector (N x 1) mean
  mu <- apply(rets, 2, sd)
  return(mu)
}

R/semidevEstimation.R
#' @name semidevEstimation
#' @aliases semidevEstimation
#' @title Estimation of the semideviation
#' @description Function which computes the semideviation.
#' @details The argument \code{control} is a list that can supply any of the following
#' components:
#' 
#' \itemize{
#' \item \code{type} method used to compute the semideviation
#' vector, among \code{'naive'} and \code{'ewma'} where:
#' 
#' \code{'naive'} is used to compute the simple semideviation.
#' 
#' \code{'ewma'} is used to compute the exponential weighted moving average
#' semideviation. The data must be sorted from the oldest to the latest. See RiskMetrics (1996).
#' 
#' The semideviation for one stock is computed as follows. First we select the
#' returns which are smaller than the average of the past returns; we get a new
#' vector of dimension \eqn{K \times 1, K \le N}{Kx1,K<=N}. Then, the weight \eqn{w_i}{w_i}
#' for each observation at its corresponding time \eqn{t} is computed as \eqn{w
#' = \lambda^{t}}{w=lambda^t}. We obtain a \eqn{K \times 1}{Kx1} vector. The vector of
#' weights is then normalized.  Finally, the semideviation is obtained as the
#' weighted standard deviation.
#' 
#' Default: \code{type = 'naive'}.
#' 
#' \item \code{lambda} decay parameter. Default: \code{lambda = 0.94}.
#' }
#' 
#' @param rets a \eqn{(T \times N)}{(T x N)} matrix of past returns.
#' @param control control parameters (see *Details*).
#' @return A \eqn{(N \times 1)}{(N x 1)} vector of semideviations.
#' @author David Ardia, Kris Boudt and Jean-Philippe Gagnon Fleury.
#' @references
#' RiskMetrics (1996)
#' \emph{RiskMetrics Technical Document}.
#' J. P. Morgan/Reuters. 
#' \url{https://www.msci.com/documents/10199/5915b101-4206-4ba0-aee2-3449d5c7e95a} 
#' @keywords htest
#' @examples
#' # Load returns of assets or portfolios
#' data("Industry_10")
#' rets = Industry_10
#' 
#' # Naive semideviation estimation
#' semidevEstimation(rets)
#' 
#' # Naive estimation of the semideviation
#' semidevEstimation(rets, control = list(type = 'naive'))
#' 
#' # Ewma estimation of the semideviation. Default lambda = 0.94
#' semidevEstimation(rets, control = list(type = 'ewma'))
#' 
#' # Ewma estimation of the semideviation. lambda = 0.9
#' semidevEstimation(rets, control = list(type = 'ewma', lambda = 0.9))
#' @export
semidevEstimation <- function(rets, control = list()) {
  
  if (missing(rets)) {
    stop("rets is missing")
  }
  if (!is.matrix(rets)) {
    stop("rets must be a (T x N) matrix")
  }
  
  ctr <- .ctrSemidev(control)
  
  if (ctr$type[1] == "naive") {
    semiDev <- .naiveSemiDev(rets = rets)
  } else if (ctr$type[1] == "ewma") {
    semiDev <- .ewmaSemiDev(rets = rets, lambda = ctr$lambda)
  } else {
    stop("control$type is not well defined")
  }
  return(semiDev)
}

.ctrSemidev <- function(control = list()) {
  
  if (!is.list(control)) {
    stop("control must be a list")
  }
  if (length(control) == 0) {
    control <- list(type = "naive", lambda = 0.94)
  }
  nam <- names(control)
  if (!("type" %in% nam) || is.null(control$type)) {
    control$type <- c("naive", "ewma")
  }
  if (!("lambda" %in% nam) || is.null(control$lambda)) {
    control$lambda <- 0.94
  }
  return(control)
}

.naiveSemiDev <- function(rets) {
  ## Compute the naive semi-deviation INPUTs rets : matrix (T x N) returns
  ## OUTPUTs semiDev : vector (N x 1) semi-deviation
  semiDev <- .ewmaSemiDev(rets, lambda = 1)
  return(semiDev)
}

.ewmaSemiDev <- function(rets, lambda) {
  ## Compute the ewma semi-deviation INPUTs rets : matrix (T x N) returns
  ## OUTPUTs semiDev : vector (N x 1) semi-deviation
  t <- dim(rets)[1]
  n <- dim(rets)[2]
  semiDev <- vector("double", n)
  mu <- colMeans(rets)
  w <- lambda^(t:1)
  for (j in 1:n) {
    retsj <- rets[, j]
    muj <- mu[j]
    idx <- retsj < muj
    wj <- w[idx]/sum(w[idx])
    semiDev[j] <- sqrt(sum(wj * (retsj[idx] - muj)^2))
  }
  return(semiDev)
}

R/RiskPortfolios.R
#' @name RiskPortfolios
#' @docType package
#' @title RiskPortfolios: Computation of risk-based portfolios in R
#' @description \code{RiskPortfolios} (Ardia et al., 2017) is an \R package for constructing risk-based portfolios dedicated to 
#' portfolio managers and quantitative analysts. It provides a set of 
#' functionalities to build mean-variance, minimum variance, inverse-volatility weighted (Leote et al., 2012), 
#' equal-risk-contribution (Maillard et al. 2010), maximum diversification (Choueifaty and Coignard, 2008), and 
#' risk-efficient (Amenc et al., 2011) portfolios. Optimization is achieved with the \R 
#' functions \code{\link[quadprog]{solve.QP}} and \code{\link[nloptr]{slsqp}}. Long or gross constraints can be added to the optimizer. 
#' As risk-based portfolios are mainly based on covariances, the package also provides a large set of 
#' covariance matrix estimators. A simulation study relying on the package is described in Ardia et al. (2017).
#' 
#' @section Functions:
#' \itemize{
#' \item \code{\link{optimalPortfolio}}: Optimization of portfolios.
#' \item \code{\link{meanEstimation}}: Computation of expected returns.
#' \item \code{\link{covEstimation}}: Computation of covariance matrix.
#' \item \code{\link{semidevEstimation}}: Computation of semi-deviation.
#' }
#' 
#' @section Update:
#' The latest version of the package is available at \url{https://github.com/ArdiaD/RiskPortfolios}.
#' @author David Ardia, Kris Boudt and Jean-Philippe Gagnon-Fleury.
#' @note Please cite the package in publications. Use \code{citation("RiskPortfolios")}.
#' @references 
#' Amenc, N., Goltz, F., Martellini, L., Retowsky, P. (2011).
#' Efficient indexation: An alternatice to cap-weightes indices.  
#' \emph{Journal of Investment Management} \bold{9}(4), pp.1-23.
#' 
#' Ardia, D., Boudt, K. (2015). 
#' Implied expected returns and the choice of a mean-variance efficient portfolio proxy.
#' \emph{Journal of Portfolio Management} \bold{41}(4), pp.66-81. 
#' \doi{10.3905/jpm.2015.41.4.068}
#' 
#' Ardia, D., Bolliger, G., Boudt, K., Gagnon-Fleury, J.-P. (2017).  
#' The Impact of covariance misspecification in risk-based portfolios.  
#' \emph{Annals of Operations Research} \bold{254}(1-2), pp.1-16. 
#' \doi{10.1007/s10479-017-2474-7}
#' 
#' Ardia, D., Boudt, K., Gagnon-Fleury, J.-P. (2017).  
#' RiskPortfolios: Computation of risk-based portfolios in R.  
#' \emph{Journal of Open Source Software} \bold{10}(2).
#' \doi{10.21105/joss.00171}
#' 
#' Choueifaty, Y., Coignard, Y. (2008).  
#' Toward maximum diversification.
#' \emph{Journal of Portfolio Management} \bold{35}(1), pp.40-51. 
#' \doi{10.3905/JPM.2008.35.1.40}
#' 
#' Choueifaty, Y., Froidure, T., Reynier, J. (2013).  
#' Properties of the most diversified portfolio.  
#' \emph{Journal of Investment Strategies} \bold{2}(2), pp.49-70. 
#' \doi{10.21314/JOIS.2013.033}
#' 
#' Das, S., Markowitz, H., Scheid, J., Statman, M. (2010).  
#' Portfolio optimization with mental accounts.  
#' \emph{Journal of Financial and Quantitative Analysis} \bold{45}(2), pp.311-334. 
#' \doi{10.1017/S0022109010000141}
#' 
#' DeMiguel, V., Garlappi, L., Uppal, R. (2009).  
#' Optimal versus naive diversification: How inefficient is the 1/n portfolio strategy.  
#' \emph{Review of Financial Studies} \bold{22}(5), pp.1915-1953. 
#' \doi{10.1093/rfs/hhm075}
#' 
#' Fan, J., Zhang, J., Yu, K. (2012).  
#' Vast portfolio selection with gross-exposure constraints.
#' \emph{Journal of the American Statistical Association} \bold{107}(498), pp.592-606. 
#' \url{10.1080/01621459.2012.68282}
#' 
#' Maillard, S., Roncalli, T., Teiletche, J. (2010).  
#' The properties of equally weighted risk contribution portfolios.  
#' \emph{Journal of Portfolio Management} \bold{36}(4), pp.60-70. 
#' \doi{10.3905/jpm.2010.36.4.060}
#' 
#' Martellini, L. (2008).  
#' Towards the design of better equity benchmarks.
#' \emph{Journal of Portfolio Management} \bold{34}(4), Summer,pp.34-41. 
#' \doi{10.3905/jpm.2008.709978}
NULL